### PR TITLE
Add a Source column to svcat describe class/plan that indicates it's user defined

### LINKED ISSUE #2217: Add a Source column to svcat describe class/plan that indicates it's user defined
The source column should display `broker` when it has a controller reference to service catalog and otherwise should display `user`.

```
$ svcat describe class mysql
  Name:          mysql1
  Description:   Helm Chart for mysql
  UUID:          mysql
  Status:        Active
  Tags:          mysql, database, sql
  Broker:        minibroker
  Source:        broker

$ svcat describe plan mysql-devdb
  Name:          mysql-devdb
  Description:   Helm Chart for mysql
  UUID:          mysql-devdb
  Status:        Active
  Tags:          mysql, database, sql
  Broker:        minibroker
  Source:        user
```

We have the following function to tell if it is a user-defined resource or managed by service catalog

https://github.com/kubernetes-incubator/service-catalog/blob/6dfe967d8041ef707af9446eddff8b2de944ee88/pkg/controller/controller_clusterservicebroker.go#L774-L781

This function needs to be moved someplace reusable by both the server and client. I think `pkg/apis/servicecatalog/v1beta1` may be a good place for it, to avoid circular package dependencies.

Tests to update:
* https://github.com/kubernetes-incubator/service-catalog/blob/6dfe967d8041ef707af9446eddff8b2de944ee88/cmd/svcat/svcat_test.go#L199-L200
* https://github.com/kubernetes-incubator/service-catalog/blob/6dfe967d8041ef707af9446eddff8b2de944ee88/cmd/svcat/svcat_test.go#L218-L220

You will need to modify the testdata to add a plan and class that are user managed (i.e. it doesn't have a controller reference on its metadata). The testdata resides here: https://github.com/kubernetes-incubator/service-catalog/tree/6dfe967d8041ef707af9446eddff8b2de944ee88/cmd/svcat/testdata/responses/catalog

### PR DESCRIPTION
**Pull request by shouhong**
This PR is a 
 - [X] Feature Implementation
 - [ ] Bug Fix
 - [ ] Documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** 

Fixes #2217

Please leave this checklist in the PR comment so that maintainers can ensure a good PR.

Merge Checklist:
 - [ ] New feature 
   - [ ] Tests
   - [ ] Documentation
 - [ ] SVCat CLI flag
 - [ ] Server Flag for config
   - [ ] Chart changes
   - [ ] removing a flag by marking deprecated and hiding to avoid
         breaking the chart release and existing clients who provide a
         flag that will get an error when they try to update

### CODE REVIEW

File: pkg/apis/servicecatalog/v1beta1/plan.go [hunk 1]
```diff
@@ -96,6 +96,16 @@ func (p *ServicePlan) GetClassID() string {
 	return p.Spec.ServiceClassRef.Name
 }
 
+// IsManaged returns whether the plan is managed or not.
+func (p *ClusterServicePlan) IsManaged() bool {
+	return IsServiceCatalogManagedResource(p.ObjectMeta.GetOwnerReferences())
```
Comment (carolynvs) [inline pkg/apis/servicecatalog/v1beta1/plan.go]: Same comment as for `class.IsManaged()`

File: pkg/apis/servicecatalog/v1beta1/plan.go [hunk 2]
```diff
@@ -96,6 +96,16 @@ func (p *ServicePlan) GetClassID() string {
...
+}
+
+// IsManaged returns whether the plan is managed or not.
+func (p *ServicePlan) IsManaged() bool {
+	return IsServiceCatalogManagedResource(p.ObjectMeta.GetOwnerReferences())
```
Comment (carolynvs) [inline pkg/apis/servicecatalog/v1beta1/plan.go]: Same as above.

**Comment by shouhong** [comment]
@carolynvs Thanks for your detailed review feedback. I have updated the code as you suggested. Can you please help review again?

**Comment by jberkhahn** [comment]
Sorry about the lack of response, Carolyn's been a bit busy.

The changes look good but it looks like you're in need of a rebase. Any chance of that?

**Comment by jberkhahn** [comment]
Due to lack of activity, we're closing this. If you still have an interest in this, please reopen with these changes.

**Pull request closed without merging**